src/expr/src/sig/agg.rs

```diff
@@ -95,37 +102,21 @@ pub unsafe fn _register(desc: AggFuncSig) {
 
 /// The global registry of function signatures on initialization.
 ///
-/// `#[function]` macro will generate a `#[ctor]` function to register the signature into this
+/// `#[aggregate]` macro will generate a `#[ctor]` function to register the signature into this
 /// vector. The calls are guaranteed to be sequential. The vector will be drained and moved into
-/// `FUNC_SIG_MAP` on the first access of `FUNC_SIG_MAP`.
+/// `AGG_FUNC_SIG_MAP` on the first access of `AGG_FUNC_SIG_MAP`.
 static mut AGG_FUNC_SIG_MAP_INIT: Vec<AggFuncSig> = Vec::new();
 
 /// Infer the return type for the given agg call.
+///
 /// Returns `None` if not supported or the arguments are invalid.
 pub fn infer_return_type(agg_kind: AggKind, inputs: &[DataType]) -> Option<DataType> {
     // The function signatures are aligned with postgres, see
     // https://www.postgresql.org/docs/current/functions-aggregate.html.
-    let return_type = match (agg_kind, inputs) {
-        // Min, Max, FirstValue, BitAnd, BitOr, BitXor
-        (
-            AggKind::Min
-            | AggKind::Max
-            | AggKind::FirstValue
-            | AggKind::BitAnd
-            | AggKind::BitOr
-            | AggKind::BitXor,
-            [input],
-        ) => input.clone(),
-        (
-            AggKind::Min
-            | AggKind::Max
-            | AggKind::FirstValue
-            | AggKind::BitAnd
-            | AggKind::BitOr
-            | AggKind::BitXor,
-            _,
-        ) => return None,
-        // Avg
+    Some(match (agg_kind, inputs) {
+        // XXX: some special cases that can not be handled by signature map.
```

Comment:
May be better to clarify what cases are not able to be handled by signature map

Reply:
Added. I also moved this function to the frontend.